### PR TITLE
Code action resolve activation updates

### PR DESCRIPTION
**Pull request by mezarin**

### CODE REVIEW

File: jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/ls/JakartaLanguageServer.java [hunk 1]
```diff
@@ -35,73 +37,145 @@
...
     @Override
     public CompletableFuture<Object> shutdown() {
-        // when shutting down LS, TextDocumentService.didClose() may not be called
-        // properly, need to clear existing diagnostics
+        // Perform some clean up. During shutdown, TextDocumentService.didClose() may not be called properly.
         ((JakartaTextDocumentService) textDocumentService).cleanDiagnostics();
+
+        // If requested by the client, on shutdown (i.e. last file closed), shutdown the language server.
+        if (capabilityManager.getClientCapabilities().shouldLanguageServerExitOnShutdown()) {
+            LOGGER.info("Jakarta EE server is shutding down");
```
Comment (mrglavas) [inline jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/ls/JakartaLanguageServer.java:144]: Minor typo. "shutding down" -> "shutting down".

File: jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/ls/JakartaLanguageServer.java [hunk 2]
```diff
@@ -35,73 +37,145 @@
...
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public CompletableFuture<Object> shutdown() {
-        // when shutting down LS, TextDocumentService.didClose() may not be called
-        // properly, need to clear existing diagnostics
+        // Perform some clean up. During shutdown, TextDocumentService.didClose() may not be called properly.
         ((JakartaTextDocumentService) textDocumentService).cleanDiagnostics();
```
[the remark this answers is not in the document]
Comment (mezarin) [reply jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/ls/JakartaLanguageServer.java]: Updated.

File: jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/settings/capabilities/ClientCapabilitiesWrapper.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+     * workspace edits for code actions, and false otherwise.
+     *
+     * Both of these feature must be present in order to implement code action
+     * resolve effectively.
+     *
+     * @return true if the client supports both code action data and resolving
+     *         workspace edits for code actions, and false otherwise
+     */
+    public boolean isCodeActionResolveSupported() {
+        return capabilities.getTextDocument() != null && capabilities.getTextDocument().getCodeAction() != null
```
[the remark this answers is not in the document]
Comment (mezarin) [reply jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/settings/capabilities/ClientCapabilitiesWrapper.java:107]: Yeah i agree, but we do want to keep it in sync for now.

[the remark this answers is not in the document]
Comment (turkeylurkey) [reply jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/settings/capabilities/ClientCapabilitiesWrapper.java:107]: ok

**Pull request merged**